OptionSelector/OSCategoriesTableViewController.m

```diff
@@ -0,0 +1,125 @@
+//
+//  OSCategoriesTableViewController.m
+//  OptionSelector
+//
+//  Created by Daniel Distant on 8/11/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "OSCategoriesTableViewController.h"
+#import "CQCategory.h"
+#import "OSStuffTableViewController.h"
+
+@interface OSCategoriesTableViewController () <UITableViewDelegate>
+
+@property (nonatomic) NSArray *CQCategories;
+
+
+
+@end
+
+@implementation OSCategoriesTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    CQCategory *dogs = [[CQCategory alloc] init];
+    CQCategory *cats = [[CQCategory alloc] init];
+    CQCategory *birds = [[CQCategory alloc] init];
+    CQCategory *mikes = [[CQCategory alloc] init];
+    
+    self.CQCategories = [[NSArray alloc] init];
+    
+    self.CQCategories = @[
+                          dogs,
+                          cats,
+                          birds,
+                          mikes
+                          ];
+    
+    dogs.name = @"dogs";
+    cats.name = @"cats";
+    birds.name = @"birds";
+    mikes.name = @"mikes";
+    
+    
+    dogs.options = @[
+                     @"siberian husky",
+                     @"pit bull",
+                     @"staffordshire terrier",
+                     @"shiba inu"
+                     ];
+    
+    //any dog under 15 lbs is a cat
+    
+    cats.options = @[
+                     @"garfield",
+                     @"chihuahua",
+                     @"poodle",
+                     @"the weird hairless kind"
+                     ];
+    
+    birds.options = @[
+                      @"peregrine falcon",
+                      @"osprey",
+                      @"golden eagle",
+                      @"red-tailed hawk named Tobias"
+                      ];
+    
+    mikes.options = @[
+                      @"default Mike with sidekick Max",
+                      @"evil French twin Mikael",
+                      @"git legend Mike with dual-wield Macbooks",
+                      @"Mike with weaponized flip-flops",
+                      @"rare holographic non-mustache Mike"
+                      ];
+
+    
+}
+
+-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    
+    OSStuffTableViewController *stuffTableViewController = [[OSStuffTableViewController alloc]init];
+    stuffTableViewController = segue.destinationViewController;
+    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+    stuffTableViewController.category = [self.CQCategories objectAtIndex:indexPath.row];
+    
+    if (([stuffTableViewController.tableView cellForRowAtIndexPath:indexPath].accessoryType = UITableViewCellAccessoryCheckmark)) {
```

Comment:
I'm seeing a few different problems here.

First, you're making the assumption that `OSStuffTableViewController` will always use a table view to display a category's options. What if you wanted to change its implementation to use a collection view, for example? You'd not only have to change its .m file, but you'd have to update this logic as well. `OSCategoriesTableViewController` should know as little about `OSStuffTableViewController` as possible. In this case, there isn't really any reason why it should know about the table view—it just needs to pass the current category along.

Second, you're using an index path from the current view controller's table view to access a cell in another table view. This is a recipe for potential crashes or other strange behavior.

Third, you're relying on a visual element (in this case the cell's `accessoryType`) to determine which category option was selected. What if you wanted to change your cells' selected state to use a different property (other than the checkmark accessory type)? You'd have to entirely rewrite this logic. A better way to handle this would be to have a property on `OSStuffTableViewController` to pass the selected category option to the master view controller.